packages/devtools_app/lib/src/flutter/notifications.dart

```diff
@@ -0,0 +1,216 @@
+    return oldWidget.data != data;
+  }
+}
+
+class NotificationsState extends State<Notifications>
+    implements NotificationProvider {
+  OverlayEntry _overlayEntry;
+
+  final List<_Notification> _notifications = [];
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+
+    if (_overlayEntry == null) {
+      _overlayEntry = OverlayEntry(
+        maintainState: true,
+        opaque: false,
+        builder: _buildOverlay,
+      );
+      SchedulerBinding.instance.scheduleFrameCallback((_) {
+        Overlay.of(context).insert(_overlayEntry);
+      });
+    }
+  }
+
+  @override
+  void dispose() {
+    _overlayEntry.remove();
+    super.dispose();
+  }
+
+  /// Pushes a notification [message].
+  @override
+  void push(String message) {
+    setState(() {
+      _notifications.add(
+        _Notification(
+          message: message,
+          remove: _removeNotification,
+        ),
+      );
+      _overlayEntry?.markNeedsBuild();
+    });
+  }
+
+  void _removeNotification(_Notification notification) {
+    setState(() {
+      final didRemove = _notifications.remove(notification);
+      if (didRemove) {
+        _overlayEntry?.markNeedsBuild();
+      }
+    });
+  }
+
+  Widget _buildOverlay(BuildContext context) {
+    return Align(
+      alignment: Alignment.bottomRight,
+      child: SizedBox(
+        width: _notificationWidth,
+        child: SingleChildScrollView(
+          reverse: true,
+          scrollDirection: Axis.vertical,
+          child: Column(
+            verticalDirection: VerticalDirection.down,
+            mainAxisSize: MainAxisSize.min,
+            children: _notifications,
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return _InheritedNotifications(data: this, child: widget.child);
+  }
+}
+
+class _Notification extends StatefulWidget {
+  const _Notification(
+      {Key key,
+      @required this.message,
+      this.callback = _noop,
+      this.duration = _defaultDuration,
+      @required this.remove})
+      : assert(message != null),
+        assert(callback != null),
+        assert(remove != null),
+        assert(duration != null),
+        super(key: key);
+
+  static const Duration _defaultDuration = Duration(seconds: 7);
+
+  final Duration duration;
+  final VoidCallback callback;
+  final String message;
+  final void Function(_Notification) remove;
+
+  static void _noop() {}
+
+  @override
+  _NotificationState createState() => _NotificationState();
+}
+
+class _NotificationState extends State<_Notification>
+    with SingleTickerProviderStateMixin {
+  AnimationController controller;
+  CurvedAnimation curve;
+  Timer _dismissTimer;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(
+        duration: const Duration(milliseconds: 400), vsync: this);
```

Comment:
we need to define a central set of animation times to use as our animation theme for the app. 400ms seems a bit long but maybe it is fine.

Reply:
Since we're refactoring things this month: https://github.com/flutter/devtools/pull/1550